arctic/store/_ndarray_store.py

```diff
@@ -467,9 +597,48 @@ def _segment_index(self, new_data, existing_index, start, new_segments):
         segments: list of offsets. Each offset is the row index of the
                   the last row of a particular chunk relative to the start of the _original_ item.
                   array(new_data) - segments = array(offsets in item)
-        
+
         Returns:
         --------
         Library specific index metadata to be stored in the version document.
         """
         pass  # numpy arrays have no index
+
+    def _chunked_update(self, arctic_lib, version, symbol, records, ranges, orig_ranges):
+        collection = arctic_lib.get_top_level_collection()
+
+        chunks = [r.tostring() for r in records]
+        lens = [len(i) for i in chunks]
+        chunks = compress_array(chunks)
+
+        seg_count = 0
+        seg_len = 0
+
+        bulk = collection.initialize_unordered_bulk_op()
+        for chunk, rng, orig_rng, rec_len in zip(chunks, ranges, orig_ranges, lens):
+            start = rng[0]
+            end = rng[1]
+            orig_start = orig_rng[0]
+            if orig_start is None:
+                version['up_to'] += rec_len
+                seg_count += 1
+                seg_len += rec_len
+            segment = {'data': Binary(chunk), 'compressed': True}
+            segment['segment'] = start
+            segment['end'] = end
+            sha = checksum(symbol, segment)
+            segment['sha'] = sha
+            if orig_start is None:
+                # new chunk
+                bulk.find({'symbol': symbol, 'sha': sha, 'segment': segment['segment']}
+                          ).upsert().update_one({'$set': segment, '$addToSet': {'parent': version['_id']}})
+            else:
+                bulk.find({'symbol': symbol, 'segment': orig_start}
+                          ).update_one({'$set': segment, '$addToSet': {'parent': version['_id']}})
```

Comment:
All the chunks in arctic (currently) have a parent field that points back up to the version document(s) that refer to the chunk.

This is somewhat backwards and exists as a result of the experiments we did when trying to figure out how to maximise read performance. Querying based on an indexed parent field was much faster than querying a set of chunks, for example. So in this case we would need to write a new version doc, and reference that version in you new chunks + all the existing chunks  (- the chunk you're replacing).  This would be as expensive as a `write()`.

Ideally it would be the other way round: we would write a new version document, and a new chunk, and be able to efficiently query the existing and new chunk using the new version document (I think this might be possible doing something clever with 2d indexes to pull out the latest chunk version across some date range).  This might not be easy without some serious playing...